src/mod/ident.mod/ident.c

```diff
@@ -116,14 +122,73 @@ static void ident_oidentd()
     putlog(LOG_MISC, "*", "Ident error: path too long.");
     return;
   }
-  if ((fd = open(path, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR | S_IROTH)) < 0) {
-    putlog(LOG_MISC, "*", "Ident error: %s", strerror(errno));
-    return;
+  fd = fopen(path, "r");
+  if (fd != NULL) {
+    /* Calculate file size for buffer */
+    if (fseek(fd, 0, SEEK_END) == 0) {
+      filesize = ftell(fd);
+      if (filesize == -1) {
+        putlog(LOG_MISC, "*", "IDENT: Error reading oident.conf");
+      }
+      data = nmalloc(sizeof(char) * (filesize + 256)); /* Room for Eggdrop adds */
+
+      /* Read the file into buffer */
+      if (fseek(fd, 0, SEEK_SET) != 0) {
+        putlog(LOG_MISC, "*", "IDENT: Error setting oident.conf file pointer");
+      }
+      while(fgets(line, 255, fd)) {
+        /* If it is not an Eggdrop entry, don't mess with it */
+        if (!strstr(line, "### eggdrop_")) {
+          strncat(data, line, ((filesize + 256) - strlen(data)));
+        } else {
+          /* If it is Eggdrop but not me, check for expiration and remove */
+          if (!strstr(line, identstr)) {
+            strncpy(buf, line, sizeof buf);
+            strtok(buf, "!");
+            prevtime = atoi(strtok(NULL, "!"));
+            if ((time(NULL) - prevtime) > 300) {
+              putlog(LOG_DEBUG, "*", "IDENT: Removing expired oident.conf entry: \"%s\"", buf);
+            } else {
+              strncat(data, line, ((filesize + 256) - strlen(data)));
+            }
+          }
+        }
+      }
+    }
+    fclose(fd);
+  } else {
+    putlog(LOG_MISC, "*", "IDENT: Error opening oident.conf for reading");
+  }
+  servidx = findanyidx(serv);
+  addr.addrlen = sizeof addr.addr.sa;
```

Comment:
In our case, getsockname() should not read but write to addr.addrlen, so i dont think we need this line.

Reply:
My comment was wrong here, and the cause of my confusion. Didnt see that getsockname() also reads from addr.addrlen. Geo is correcting the value right now... thx thommey.